redwood-lazylayout-view/src/main/kotlin/app/cash/redwood/lazylayout/view/ViewLazyList.kt

```diff
@@ -107,7 +107,7 @@ internal open class ViewLazyList private constructor(
             val viewsOnScreen = recyclerView.children.map { recyclerView.getChildViewHolder(it).bindingAdapterPosition }
             userHasScrolled = true // Prevent guest code from hijacking the scrollbar.
             onViewportChanged?.invoke(
-              viewsOnScreen.min(),
+              viewsOnScreen.min().coerceAtLeast(0),
```

Comment:
`.min()` _sometimes_ returns -1 (which had been observed / is expected)

Reply:
-1 is `NO_POSITION` which is what is returned for all positions between adapter invalidations and the next layout pass.

Also `min` and `max` throw if the sequence is empty, are we assuming the scroll callback will never be invoked on an empty list?

Reply:
> Also min and max throw if the sequence is empty, are we assuming the scroll callback will never be invoked on an empty list?

ahhh 🤔, that's the case (not invoked on empty list) so far (this has only only been applied to emoji search). But it's not guaranteed the invocation wouldn't happen on an empty list (someone else might use it, or other teams might use it).